src/components/StepIndicator/__tests__/StepIndicator.spec.js

```diff
@@ -4,40 +4,45 @@
  */
 
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render } from '@testing-library/react';
 
-import { Step, StepIndicator } from '../';
+import { Step, StepIndicator } from '../../..';
 
 describe('StepIndicator', () => {
-  let list;
-
-  beforeEach(() => {
-    list = shallow(
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
       <StepIndicator currentIndex={1}>
-        <Step
-          label="First step"
-          description="Step 1: Getting started with Carbon Design System"
-        />
-        <Step
-          label="Second step"
-          description="Step 2: Getting started with Carbon Design System"
-        />
-      </StepIndicator>
+        <Step label="test label 1" description="test description 1" />
+        <Step label="test label 2" description="test description 2" />
+      </StepIndicator>,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
     );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('StepIndicator');
   });
 
-  describe('render', () => {
-    it('renders correctly', () => {
-      expect(list).toMatchSnapshot();
-    });
+  test('should add a custom class to the parent component', () => {
+    const { container } = render(<StepIndicator className="custom-class" />);
+    expect(container.firstElementChild).toHaveClass('custom-class');
+  });
 
-    it("renders the HTML of the node's subtree", () => {
-      expect(list.render()).toMatchSnapshot();
-    });
+  test('should add a custom class to the child step component', () => {
+    const { getByText } = render(
+      <StepIndicator>
+        <Step label="test label" className="custom-step-class" />
+      </StepIndicator>
+    );
+    expect(getByText(/test label/i).closest('li')).toHaveClass(
+      'custom-step-class'
+    );
+  });
 
-    it('renders if `currentIndex` is null', () => {
-      list.setProps({ currentIndex: null });
-      expect(list.render()).toMatchSnapshot();
-    });
```

Comment:
I don't think this test is necessary, as it is built-in to the underlying `ProgressIndicator`